src/service.c

```diff
@@ -199,6 +202,17 @@ device_compare_func (gconstpointer ga, gconstpointer gb)
         }
     }
 
+  technology = UP_DEVICE_TECHNOLOGY_UNKNOWN;
+  if (!ret && (a_technology != b_technology) &&
+      (a_technology == technology || b_technology == technology))
+    {
+      if (a_technology != technology) { /* a has a defined battery technology */
```

Comment:
And here you can just do
```suggestion
      if (a_technology > b_technology) {
```
so, if `a_technology` is unknown, it will be sorted last.

Reply:
Not sure if this change makes sense if the previous one isn't made?